### ISSUE TITLE
Adding followthemoney schemas

### ISSUE THREAD

**Issue by winers1290** [alephdata/aleph#563]
So I've been trying to edit to followthemoney schemas in a development install locally to try and add these 'cyber entities'.

I'm executing bash within the image: `docker exec -it "58b1779d4915" bash`, where I add the extra files. When I executed the `aleph` command, I could see that it picked up the changes, as I made a typo once and it made an exception.

I've made a test CSV and defined my YAML. When I upload the file it processes successfully, and the entry gets populated in the front end, but none of my line items within the CSV are populated, the collection is blank and showing zero entries... Here is my test YAML:

````
test:
  label: Accounts Info
  info_url: https://google.com
  category: leak
  summary: This is a dataset
  roles:
    - guest
  queries:
    - csv_url: http://192.168.0.112:8000/users.csv
      entities:
        account:
          schema: OnlineIdentity
          keys:
            - email
            - account
            - ip
          properties:
            account:
              column: account
            email:
              column: email
            ip:
              column: ip
            sourceUrl:
              column: sourceUrl
            plainPassword:
              colum: plainPassword
            hashedPassword:
              column: hashedPassword
````

Any ideas? 

---

UPDATE: Tested this with a factory default schema, and seeing the same problem, no records get uploaded. I've tested using both the `aleph bulkload` command and the client side `alephclient` command over API, and both have the same result.

**Comment by pudo** [comment]
Regarding extending ``followthemoney``: I've been thinking about this a bit. My proposed solution would be to introduce an environment variable, say ``FTM_MODEL_PATH`` (hooked in [here](https://github.com/alephdata/followthemoney/blob/master/followthemoney/__init__.py#L9)) that can be used to override the schema set without actually modifying the source code of any of the tool. Does that make sense to you? 

Regarding the collection, a quick note: the counts in a collection get cached for an hour. All collection metadata is re-indexed once an hour in production. If you wish to manually fix this, you can run ``aleph repair`` within an aleph container. (Background: it turns out running the count queries on the data is a little intense on the ES cluster, so I'm trying to do it systematically, not ad-hoc.)

**Comment by winers1290** [comment]
Adding the environment variable makes sense to me, we'll just run a check to see if it's defined, and if not load as per normal.

However, this doesn't explain the issue I'm currently having. Are their issues with the most recent release? No matter how I go about loading CSV files, the data from the file does not show up inside Aleph, even though the collection gets parsed fine and their are no reported issues with my YAML file.

I've ran `aleph repair` and everything gets indexed, but there are just no items within these collections. I'm positive that the CSV file gets read, as when I'm using `aleph client` I can `curl` it and see it read in the web logs, and when I load it directly in the Docker image there doesn't seem to be any issues, and when I give it larger CSVs the processing time increases.

I'm just using the most current release, following the instructions for the development installation. Should I be using one of the stable releases for this?

**Comment by winers1290** [comment]
Here is the output from when I run the `aleph bulkload` command, using a sample from the Experian breach.

````
INFO:aleph:Loading bulk data from: /host/home/aleph/experian.yml
INFO:followthemoney.mapping.csv:Loading: http://192.168.0.112:8000/Experian_Guam.csv
INFO:celery.app.trace:Task aleph.logic.entities.bulk.bulk_load_query[35b99771-f111-4919-bd1c-f55513cc6355] succeeded in 0.061473847017623484s: None
````

**Comment by pudo** [comment]
Can you check that either a) in the environment you're running this in, the env var ``ALEPH_EAGER`` is set to ``true``, or b) that you have at least one aleph worker service running?

**Comment by winers1290** [comment]
I've checked my environment variables using `printenv` and do not have a variable named `ALEPH_EAGER`. Should I define one?

How do I check if the aleph worker service is running>

**Comment by pudo** [comment]
Yes, try setting ``ALEPH_EAGER`` to ``true`` inside a docker shell for testing. 

You can check if an aleph docker worker is running by running the following in the base directory outside of docker:

``docker-compose ps | grep worker``

**Comment by pudo** [comment]
To clarify: what ``ALEPH_EAGER`` does is to disable the delegation of tasks (such as bulk loading data) from the shell to a worker node. This is good for testing, but not ideal in a production scenario where you want tasks to be sent to an available worker.

**Comment by winers1290** [comment]
Okay, set the env. But looks like the lack of a worker might be the issue, as it's not showing up.

Here is the ungrepped list of my docker processes:

````

aleph_api_1                python3 aleph/manage.py ru ...   Up               0.0.0.0:5000->5000/tcp
aleph_convert-document_1   python3 convert/server.py        Up (healthy)     3000/tcp
aleph_elasticsearch_1      /usr/local/bin/docker-entr ...   Up               9200/tcp, 9300/tcp
aleph_extract-entities_1   python3 /service/entityext ...   Up (healthy)     50000/tcp
aleph_postgres_1           docker-entrypoint.sh postgres    Up               0.0.0.0:15432->5432/tcp
aleph_rabbitmq_1           docker-entrypoint.sh rabbi ...   Up               25672/tcp, 4369/tcp, 5671/tcp, 5672/tcp
aleph_recognize-text_1     python3 /service/textrecog ...   Up (unhealthy)   50000/tcp
aleph_redis_1              docker-entrypoint.sh redis ...   Up               6379/tcp
aleph_ui_1                 npm run start                    Up               0.0.0.0:8080->3000/
````

Do I need to start the worker manually somehow?

**Comment by winers1290** [comment]
Cool, just emailed you.

**Comment by pudo** [comment]
OK, so the support for custom model directories in followthemoney is now published as version 1.4.0. If you wish to use this in production you probably still have to derive a docker image with the additional YAML files.
